.github/workflows/containerized-ci.yml

```diff
@@ -1,84 +1,123 @@
 name: Continuous integration in a box
-on: 
-  push: 
-    branches-ignore:    
-      - documentation
-  pull_request: 
-    branches-ignore:    
-      - documentation
+on:
+  push:
+    branches-ignore:
+    - documentation
+  pull_request:
+    branches-ignore:
+    - documentation
+
+defaults:
+  run:
+    shell: bash
 
 jobs:
   Containerized-CI:
     runs-on: ubuntu-20.04
     strategy:
       fail-fast: false
       matrix:
+        fortran-compiler: [ifort, nvfortran]
         rte-kernels: [default, openacc]
-        container: ["earthsystemradiation/rte-rrtmgp-ci:ifort","earthsystemradiation/rte-rrtmgp-ci:nvfortran"]
+        include:
+        - fortran-compiler: ifort
+          fcflags: "-m64 -g  -traceback -heap-arrays -assume realloc_lhs -extend-source 132 -check bounds,uninit,pointers,stack -stand f08"
+          image: "earthsystemradiation/rte-rrtmgp-ci:ifort"
+        - fortran-compiler: nvfortran
+          fcflags: "-Mallocatable=03 -Mstandard -Mbounds -Mchkptr -Kieee -Mchkstk"
+          image: "earthsystemradiation/rte-rrtmgp-ci:nvfortran"
+        - netcdf-c: /home/runner/netcdf-c
+        - netcdf-fortran: /home/runner/netcdf-fortran
     container:
-      image: ${{ matrix.container }}
+      image: ${{ matrix.image }}
     env:
-      NCHOME: /home/runner/netcdf-c
-      NFHOME: /home/runner/netcdf-fortran
-      RFMIP_DIR: /home/runner/rfmip-files
+      # Core variables:
+      FC: ${{ matrix.fortran-compiler }}
+      FCFLAGS: ${{ matrix.fcflags }}
+      # Make variables:
+      NCHOME: ${{ matrix.netcdf-c }}
```

Comment:
@skosukhin Can you explain why it's preferable to use environment variables from the `matrix` rather than setting them within the job?

Reply:
I'm sorry, I see two questions here:

1. Why do I set environment variables for all steps of a job and not only for steps that really need them?
    
    I simply prefer having all environment variables set in one place. Of course, a counter-argument to that is that we might miss certain side effects for steps that should not need the variables but implicitly use them.

2. Why do I set variables in the matrix and then re-use them as is to set environment variables?
    
    I consider variables in the matrix as input parameters for a job and environment variables as something that enables it. For example, the build system in `develop` is designed and tested with the requirement that `FCFLAGS`, `NFHOME`, etc. are set in the environment. The `autoconf` build system, however, does not need it: the flags and the path to NetCDF Fortran are supposed to be provided on the command line when calling the configure script. I now realise that my comments are misleading and I also should try to get rid of `RFMIP_CACHEDIR` to make it consistent.